code/__DEFINES/movement.dm

```diff
@@ -0,0 +1,35 @@
+//The minimum for glide_size to be clamped to.
+#define MIN_GLIDE_SIZE 1
+//The maximum for glide_size to be clamped to.
+//This shouldn't be higher than the icon size, and generally you shouldn't be changing this, but it's here just in case.
+#define MAX_GLIDE_SIZE 32
+
+// Originally a really stupid /tg/ var that sucked and was really bad and caused it to look horrible. Now it's a way of compensating for time dilation
+GLOBAL_VAR_INIT(glide_size_multiplier, 1.0)
+
+// Credit to the Goon coder MBC for figuring this out!
+//Define clientside tick lag seperately from world.tick_lag
+//'cause smoothness looks good.
+// Glides are supposed to automatically adjust to client framerate. HOWEVER THEY DO NOT :: http://www.byond.com/forum/?post=2241289
+// We do the glide size compensation manually in the relevant places.
+#define CLIENTSIDE_TICK_LAG_SMOOTH 0.25
+//fuck me, I have no idea why there's only 2 framerates that handle smooth glides for us. It's probably because byond is bugged.
+//anyway just putting this define here for the client framerate toggle button between SMOOTH AND CHUNKY OH YEAH
+#define CLIENTSIDE_TICK_LAG_CHUNKY 0.4
+
+//its the future now
+#define CLIENTSIDE_TICK_LAG_CREAMY 0.15
+
+//tg coder movement speeds
+// 125 fps
+#define CLIENTSIDE_TICK_LAG_SILKY 0.08
+// 10 fps, for when you want to travel back to 2012 /tg/station.
+#define CLIENTSIDE_TICK_LAG_CLASSIC 1
+
+///Broken down, here's what this does:
+/// divides the world icon_size (32) by delay divided by the chosen ticklag from the 3 options to get the number of pixels something should be moving each tick.
+/// The division result is given a min value of 1 to prevent obscenely slow glide sizes from being set
+/// Then that's multiplied by the global glide size multiplier. 1.25 by default feels pretty close to spot on. This is just to try to get byond to behave.
+/// The whole result is then clamped to within the range above.
+/// Not very readable but it works
+#define DELAY_TO_GLIDE_SIZE(delay) (CLAMP(((32 / max((delay) / world.tick_lag, 1)) * GLOB.glide_size_multiplier), MIN_GLIDE_SIZE, MAX_GLIDE_SIZE))
```

Comment:
does the glidesize multipler have to be a GLOB, or can it just be a define or raw value

Reply:
i can axe it, its a leftover from some old code

Reply:
fixed